### PR TITLE
Aea 628 improve request validator

### PR DESCRIPTION
**Pull request by conner-jowitt**

### CODE REVIEW

File: coordinator/src/models/errors/validation-errors.ts [hunk 1]
```diff
@@ -3,77 +3,33 @@ import {MessageType} from "../../routes/util"
...
+  expression: Array<string>
 
   constructor(fieldName: string, uniqueFieldValues: Array<string>) {
-    super()
     this.message = `Expected all MedicationRequests to have the same value for ${
       fieldName
     }. Received ${[
       ...uniqueFieldValues
     ]}.`
+    this.expression = [`Bundle.entry.resource.ofType(MedicationRequests).${fieldName.split(" ")[0]}`]
```
Comment (dan-jones-iw) [inline coordinator/src/models/errors/validation-errors.ts:22]: fix to look like validator

File: coordinator/src/routes/util.ts [hunk 2]
```diff
@@ -50,10 +50,10 @@ export function identifyMessageType(bundle: fhir.Bundle): string {
   return getMessageHeader(bundle).eventCoding?.code
 }
 
-export function validatingHandler(requireSignature: boolean, handler: Handler<fhir.Bundle>) {
+export function validatingHandler(handler: Handler<fhir.Bundle>) {
   return async (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit): Promise<Hapi.ResponseObject> => {
     const requestPayload = getPayload(request)
```
Comment (dan-jones-iw) [inline coordinator/src/routes/util.ts:55]: `as fhir.Bundle` here

File: coordinator/src/models/errors/validation-errors.ts (hunk 1 above)
Comment (conner-jowitt) [reply coordinator/src/models/errors/validation-errors.ts:22]: Had a call around this, it was decided it wasn't important

File: coordinator/src/routes/util.ts [hunk 3]
```diff
@@ -83,14 +82,8 @@ function toFhirError(validation: Array<errors.ValidationError>): fhir.OperationO
   const mapValidationErrorToOperationOutcomeIssue = (ve: errors.ValidationError) => ({
     severity: ve.severity,
     code: ve.operationOutcomeCode,
-    details: {
```
Comment (mfjarvis) [inline coordinator/src/routes/util.ts:86]: I think we should talk to Kevin about whether this coding is required

File: coordinator/src/models/errors/validation-errors.ts (hunk 1 above)
Comment (mfjarvis) [inline coordinator/src/models/errors/validation-errors.ts:22]: Might be a good idea to use the field name as passed in. Then the field name could be something like:
```
Bundle.entry.resource.ofType(MedicationRequest).extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType")
```

Comment (mfjarvis) [reply coordinator/src/models/errors/validation-errors.ts:22]: Or at least change the field names to valid FHIR expressions, e.g.:
```
extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-performerSiteType")
```

Comment (mfjarvis) [reply coordinator/src/models/errors/validation-errors.ts:22]: (I know I wrote the field names in the first place :shushing_face:)

File: coordinator/src/services/validation/bundle-validator.ts [hunk 4]
```diff
@@ -59,102 +27,85 @@ function verifyCancellationBundle(bundle: fhir.Bundle): Array<errors.ValidationE
...
-    (medicationRequests: Array<fhir.MedicationRequest>) => verifyValueIdenticalForAllMedicationRequests(
-      medicationRequests,
-      "dispenseRequest.expectedSupplyDuration",
-      (medicationRequest) => medicationRequest.dispenseRequest.expectedSupplyDuration
-    ),
-    (medicationRequests: Array<fhir.MedicationRequest>) => verifyValueIdenticalForAllMedicationRequests(
-      medicationRequests,
-      "dispenseRequest.extension (performer site type)",
-      (medicationRequest) => getExtensionForUrl(
+  const consistencyValidators: Array<medicationRequestValidator> = [
```
Comment (mfjarvis) [inline coordinator/src/services/validation/bundle-validator.ts:36]: nice :ok_hand:

**Review by mfjarvis — DISMISSED** [review]

File: coordinator/src/services/validation/bundle-validator.ts [hunk 5]
```diff
@@ -59,102 +27,85 @@ function verifyCancellationBundle(bundle: fhir.Bundle): Array<errors.ValidationE
...
+    },
+    {
+      fieldName: 'extension("https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner")',
+      fieldAccessor: (medicationRequest) => getExtensionForUrlOrNull(
         medicationRequest.extension,
         "https://fhir.nhs.uk/R4/StructureDefinition/Extension-DM-ResponsiblePractitioner",
         "MedicationRequest.extension"
       )
-    ),
-    (medicationRequests: Array<fhir.MedicationRequest>) => verifyValueIdenticalForAllMedicationRequests(
```
Comment (mfjarvis) [inline coordinator/src/services/validation/bundle-validator.ts:131]: Although this one went missing

Comment (conner-jowitt) [reply coordinator/src/services/validation/bundle-validator.ts:131]: Not sure how that happened, I'll add it in now

**Review by mfjarvis — APPROVED** [review]

**Pull request merged**